backend/db/refresh.py

```diff
@@ -38,8 +38,13 @@ def refresh_db(
     local = use_local_database
     print('INFO: Starting database refresh.', flush=True)  # flush: for gh action
     t0, t0_str = datetime.now(), current_datetime()
-    update_db_status_var('last_refresh_request', t0_str, local)
+
+    if check_db_status_var('refresh_status') == 'active':
+        update_db_status_var('new_request_while_refreshing', t0_str, local)
```

Comment:
I think the way you've done this will work. My initial thought of doing this was:
- If refresh status is active, exit out (as you've done)
- Rather than checking a variable `new_request_while_refreshing `, at the end of the refresh, if the `last_refresh_request` has changed / is newer than what was set at the beginning of the script, it knows that a new request while refreshing has occurred, and use that information to know to start a new refresh.

Depending on time, while I'm testing this out I may refactor this to what I've described above. But I'm also happy with what you have!